scripts/tests/substaking_scenario.js

```diff
@@ -0,0 +1,85 @@
+            {action: 'claimSub', 'staker': 'binanceStaker2', stakeIdx: 0}, // claim for cleanup
+            {action: 'claim', staker: 'alice', stakeIdx: 0, assertClaimEquals: 250000000},
+            {action: 'lockRewards', staker: 'bob', stakeIdx: 0, assertStakeIncreaseEquals: 250000000},
+            {action: 'extend', staker: 'carol', stakeIdx: 0, unlockQuarter: 5, assertSharesEqual: 10**6 * 150},
+            {action: 'claim', staker: 'manager', stakeIdx: 0, assertClaimEquals: 250000000}, // claim rewards from Staking:Q1
+            {action: 'decalreRewardSub', quarterId: 1}, // declare rewards to SubStaking:Q1
+            {action: 'increaseTimeTo', quarterIdx: 3}, // Staking:Q3, SubStaking:Q2
+            // Alice shares in Q2 = 100 * (10**6) = 100,000,000
+            // Manager shares in Q2 = 100 * (10**6) = 100,000,000
+            // Bob shares in Q2 = 100 * (10**6 + 250000000) = 25100000000
+            // Carol shares in Q2 = 150 * (10**6)
+            // Total shares Q2 = 25450000000
+            // Q2 share price = 10**9 / 25450000000 = 0.0392927308447937 (BBS per share)
+            {action: 'claim', staker: 'alice', stakeIdx: 0, assertClaimEquals: 4929273}, // Q2 rewards + initial stake
+            {action: 'claim', staker: 'bob', stakeIdx: 0, assertClaimEquals: 1237247544}, // Q2 Rewards + Q1 Rewards + initial stake
+            {action: 'claim', staker: 'carol', stakeIdx: 0, assertClaimEquals: 255893909}, // Q2 Rewards + Q1 Rewards
+            {action: 'claim', staker: 'manager', stakeIdx: 0, assertClaimEquals: 3929273 + 10**6}, // Q2 Rewards + initial stake
+
+            // In this stage we are in the middle of Substaking Q2 (rewards decalred only for Q0 and Q1, and rewards can be claimed up to Q1 including)
+            {action: 'deposit', staker: 'manager', 'amount': 10**6 }, // deposit back the original stake amount before Q2 ends on SubStaking
+            {action: 'decalreRewardSub', quarterId: 2}, // declare rewards to SubStaking:Q2
+            {action: 'increaseTimeTo', quarterIdx: 4}, // Staking:Q4, SubStaking:Q3
+            {action: 'claimSub', 'staker': 'binanceStaker1', stakeIdx: 0, assertClaimEquals: 127464636}, // 127464636 = 0.5 * (10**6 (original amount) + 3929273(Q2 reward) + 250000000(Q1 reward))
+            {action: 'claimSub', 'staker': 'binanceStaker2', stakeIdx: 0, assertClaimEquals: 127464636}
+    ]);
+
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
+    await network.provider.send('evm_mine');
```

Comment:
Comment and loop, perhaps there is an argument you can give the provider to mine n blocks.

Reply:
yes there's a way to that in version 2.9.0 - https://hardhat.org/hardhat-network/reference/#hardhat-mine

Reply:
will add a comment about this option.